### DISCUSSION TITLE
Problems with QImg inside QAvatar

### DISCUSSION THREAD

**Discussion by tvogel** [quasarframework/quasar#15382] [General - Components / Directives / etc]
Even though the docs say, QImg would default to `object-fit: cover;`, the image does not cover the full extent when used inside QAvatar. I compiled a few examples here:

https://codepen.io/eijk/pen/qBygPKv

Is this a bug or am I doing something wrong?

**Comment by pdanpdan** [comment]
QAvatar tries to fit the image, so you'll get a small height image inside
Use `ratio="1"` on QImg to tell it is is square

**Reply by tvogel** [reply]
Well, avatars are always square, so why is this not the default? There is already special casing for QImg in QAvatar CSS.

**Comment by pdanpdan** [comment]
The special casing is exactly so that QImg in QAvatar is not affected
If you want QImg in QAvatar specify that you want it square

**Reply by tvogel** [reply]
It's also inconsistent behavior in comparison to using plain `<img>`. Without compelling reasons for the current behavior, I still tend to call it a bug.
